syntax = "proto3";

package rlenv;

import "history.proto";
import "pokemon.proto";
import "enums.proto";

message LegalActions {
    bool move1 = 1;
    bool move2 = 2;
    bool move3 = 3;
    bool move4 = 4;
    bool switch1 = 5;
    bool switch2 = 6;
    bool switch3 = 7;
    bool switch4 = 8;
    bool switch5 = 9;
    bool switch6 = 10;
}

message Info {
    int32 gameId = 1;
    bool done = 2;
    float playerOneReward = 3;
    float playerTwoReward = 4;
    bool playerIndex = 5;
    int32 turn = 6;
}

message Move {
    enums.MovesEnum moveId = 1;
    int32 pp = 2;
    int32 maxpp = 3;
}

message Moveset {
    Move move0 = 1;
    Move move1 = 2;
    Move move2 = 3;
    Move move3 = 4;
}

message Team {
    pokemon.Pokemon active = 1;
    pokemon.Pokemon bench0 = 2;
    pokemon.Pokemon bench1 = 3;
    pokemon.Pokemon bench2 = 4;
    pokemon.Pokemon bench3 = 5;
    pokemon.Pokemon bench4 = 6;
    pokemon.Pokemon bench5 = 7;
}

message State {
    Info info = 1;
    LegalActions legalActions = 2;
    repeated history.HistoryStep history = 3;
    Moveset moveset = 4;
    Team team = 5;
}
